org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/preferences/PersistentModel.java

```diff
@@ -11,37 +11,35 @@
 import java.util.Collection;
 import java.util.List;
 
-import com.google.common.base.Optional;
-
 import org.eclipse.core.resources.IFolder;
-import org.eclipse.core.resources.IResource;
 import org.eclipse.core.runtime.IPath;
 import org.eclipse.jdt.core.IClasspathEntry;
 
 /**
  * Interface to load and store Gradle model elements stored in the workspace plugin state area.
  * <p/>
- * The {@link Optional} return type on the getters mark whether a value is present in the model.
+ * Each getter may return {@code null} values if the model element is not available. If a
+ * {@code null} value is passed to a setter then the model element is deleted.
  *
  * @author Donat Csikos
  */
 public interface PersistentModel {
 
-    Optional<IPath> getBuildDir();
+    IPath getBuildDir();
 
-    void setBuildDir(Optional<IPath> buildDirPath);
+    void setBuildDir(IPath buildDir);
 
     Collection<IPath> getSubprojectPaths();
 
     void setSubprojectPaths(Collection<IPath> subprojectPaths);
 
-    Optional<List<IClasspathEntry>> getClasspath();
+    List<IClasspathEntry> getClasspath();
 
     void setClasspath(List<IClasspathEntry> classpath);
 
-    Collection<IResource> getDerivedResources();
+    Collection<IPath> getDerivedResources();
```

Comment:
That's just me getting confused about it. No, I was wrong, the build folder is never set to null. If it's not available in the Gradle model, we just set it to `build`.